code/datums/helper_datums/construction_datum.dm

```diff
@@ -62,6 +62,17 @@
 		qdel(holder)
 	return
 
+/datum/construction/proc/spawn_mecha_result()
+	if(result)
+		var/obj/mecha/m = new result(get_turf(holder))
+		var/obj/item/c = locate(/obj/item/stock_parts/cell) in holder
+		c.forceMove(m)
+		m.cell = c
+		if(findtext(result,"/obj/mecha/combat/"))

```

Comment:
result needs to be stringified first, I believe

Reply:
result is already a string

Reply:
There's a difference between a type and string. This looks like a type because of [this line](https://github.com/tgstation/tgstation/pull/30942/files#diff-b1dba44e1d25a22a29321794be968b19R67). In fact, the check should be `ispath(result, /obj/mecha/combat)`